tests/test_mask_creation.py

```diff
@@ -51,3 +50,28 @@ def test_create_sidecar_mask(data):
     # Check the .msk file, too.
     with rasterio.open(str(data.join('RGB.byte.tif.msk'))) as msk:
         assert (mask == msk.read(1, masked=False)).all()
+
+
+def test_create_mask_windowed_sidecar(data):
+    """Writing masks by window succeeds with sidecar mask
+    """
+    with rasterio.Env(GDAL_TIFF_INTERNAL_MASK=False):
+        with rasterio.open(str(data.join('RGB.byte.tif')), 'r+') as dst:
```

Comment:
@perrygeo here and below: shouldn't you be opening a temporary file here rather than updating our test dataset?